### PR TITLE
Another set of delinting

### PR DESCRIPTION
**Pull request by omoerbeek**
A set of random files made clean. During this process .clang-tidy.full was also amended a bit.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: .clang-tidy.full [hunk 1]
```diff
@@ -1,5 +1,9 @@
 ---
+<<<<<<< HEAD
```
Comment (rgacogne) [inline .clang-tidy.full:2]: That cannot be right :)

**Review by rgacogne — CHANGES_REQUESTED** [review]
One issue, the rest looks good to me!

File: pdns/zonemd.cc [hunk 2]
```diff
@@ -154,7 +160,7 @@ void pdns::ZoneMD::readRecord(const DNSRecord& record)
   d_resourceRecordSetTTLs[key] = record.d_ttl;
 }
 
-void pdns::ZoneMD::verify(bool& validationDone, bool& validationOK)
+void pdns::ZoneMD::verify(bool& validationDone, bool& validationOK) // NOLINT
```
Comment (fredmorcos) [inline pdns/zonemd.cc:163]: I think this NOLINT can be removed?

File: pdns/recursordist/ws-recursor.cc [hunk 3]
```diff
@@ -1402,24 +1433,24 @@ void AsyncWebServer::serveConnection(std::shared_ptr<Socket> client) const
...
-    auto handler = std::make_shared<TCPIOHandler>("", false, client->releaseHandle(), timeout, tlsCtx);
+    auto handler = std::make_shared<TCPIOHandler>("", false, socket->releaseHandle(), timeout, tlsCtx);
 
     PacketBuffer data;
     try {
       while (!req.complete) {
         auto ret = arecvtcp(data, 16384, handler, true);
         if (ret == LWResult::Result::Success) {
-          string str(reinterpret_cast<const char*>(data.data()), data.size());
+          string str(reinterpret_cast<const char*>(data.data()), data.size()); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
```
Comment (fredmorcos) [inline pdns/recursordist/ws-recursor.cc:1453]: I think a comment is important here.

File: pdns/recursordist/ws-recursor.cc [hunk 4]
```diff
@@ -1296,39 +1322,46 @@ void RecursorWebServer::jsonstat(HttpRequest* req, HttpResponse* resp)
...
-    for (const rcounts_t::value_type& q : rcounts) {
-      totIncluded -= q.first;
+    unsigned int tot = 0;
+    unsigned int totIncluded = 0;
+    for (const rcounts_t::value_type& count : rcounts) {
+      totIncluded -= count.first;
       entries.push_back(Json::array{
-        -q.first, q.second.toString()});
-      if (tot++ >= 100)
+        -count.first, count.second.toString()}); // NOLINT: union
```
Comment (fredmorcos) [inline pdns/recursordist/ws-recursor.cc:1361]: A reference to #12809 here and next to the other NOLINT union lines would be great.

**Review by fredmorcos — CHANGES_REQUESTED** [review]
I wasnt super thorough since I'm on mobile. My general idea is:
1. A complete NOLINT is not a good idea. It's better we disable specific lint's on lines.
2. A comment is almost always needed giving a quick explanation.
3. A link to the relevant issue if applicable.

Also, I hope you're using the eglot rename symbol function 😃

**Comment by omoerbeek** [comment]
> Also, I hope you're using the eglot rename symbol function 😃

I am ;-)

File: pdns/recursordist/ws-recursor.cc (hunk 3 above)
Comment (omoerbeek) [reply pdns/recursordist/ws-recursor.cc]: Done

File: pdns/recursordist/ws-recursor.cc (hunk 4 above)
Comment (omoerbeek) [reply pdns/recursordist/ws-recursor.cc]: Removed, as this is now covered by `.clang-tidy.full` changes.

**Review by fredmorcos — APPROVED** [review]

**Review by rgacogne — APPROVED** [review]

**Pull request merged**